src/Queue.js

```diff
@@ -3,88 +3,113 @@ const { SchedulerInterrupt } = require('./constants/index');
+    // The queue's priority level; the lower the number, the higher the priority
+    this.priorityLevel = priorityLevel;
+    // The queue's parent scheduler
+    this.scheduler = scheduler;
+    // The queue's allotted time slice; each process in this queue is executed for this amount of time
+    this.quantum = quantum;
+    // A counter to keep track of how much time the queue has been executing so far
+    this.quantumClock = 0;
+    this.queueType = queueType;
+  }
 
-    // Adds the input process to the queue's list of processes
-    // Also sets the input process's parent queue to this queue
-    // Return the newly added process
-    enqueue(process) {
-        
-    }
+  // Adds the input process to the queue's list of processes
+  // Also sets the input process's parent queue to this queue
+  // Return the newly added process
+  enqueue(process) {
+    this.processes.push(process);
+    process.setParentQueue(this);
+    return process;
+  }
 
-    // Removes the least-recently added process from the list of processes
-    // Return the newly-removed process
-    dequeue() {
-        
-    }
+  // Removes the least-recently added process from the list of processes
+  // Return the newly-removed process
+  dequeue() {
+    return this.processes.shift();
+  }
 
-    // Return the least-recently added process without removing it from the list of processes
-    peek() {
-        
-    }
+  // Return the least-recently added process without removing it from the list of processes
+  peek() {
+    return this.processes[0];
+  }
 
-    // Checks to see if there are any processes in the list of processes
-    isEmpty() {
+  // Checks to see if there are any processes in the list of processes
+  isEmpty() {
+    return this.processes.length === 0;
+  }
 
-    }
+  // Return this queue's priority level
+  getPriorityLevel() {
+    return this.priorityLevel;
+  }
 
-    // Return this queue's priority level
-    getPriorityLevel() {
-        
-    }
+  // Return this queue's queueType
+  getQueueType() {
+    return this.queueType;
+  }
 
-    // Return this queue's queueType
-    getQueueType() {
-        
+  // Manages a process's execution for the appropriate amount of time
+  // Checks to see if currentProcess's `this.stateChanged` property is true
+  // If it is, reset `this.quantumClock` to give the process more time and return
+  // Otherwise, increment `this.quantumClock` by `time`
+  // Check to see if `this.quantumClock` is greater than `this.quantum`
+  // If it is, then we need to execute the next process in the queue
+  // Set `this.quantumClock` to 0
+  // Dequeue the next process from the queue
+  // If it isn't finished, emit a scheduler interrupt notifying the scheduler that this process
+  // needs to be moved to a lower priority queue
+  manageTimeSlice(currentProcess, time) {
+    if (currentProcess.isStateChanged()) {
+      this.quantumClock = 0;
+      return;
+    } else {
+      this.quantumClock += time;
```

Comment:
For this one you could have removed the else statement and just did this line of code on it's own line after the if check
`this.quantumClock += times;`
again small thing and not necessary just a stylistic thing I wanted to bring up.